aws/resource_aws_db_instance.go

```diff
@@ -1470,7 +1470,8 @@ func resourceAwsDbInstanceUpdate(d *schema.ResourceData, meta interface{}) error
 	if d.HasChange("allow_major_version_upgrade") {
 		d.SetPartial("allow_major_version_upgrade")
 		req.AllowMajorVersionUpgrade = aws.Bool(d.Get("allow_major_version_upgrade").(bool))
-		requestUpdate = true
+		// Having allowing_major_version_upgrade by itself should not trigger ModifyDBInstance
+		// InvalidParameterCombination: No modifications were requested
```

Comment:
I feel the sentence is cut off and would suggest
```suggestion
		// as it results in InvalidParameterCombination: No modifications were requested
```